Flagsmith.FlagsmithClient/Mappers.cs

```diff
@@ -0,0 +1,294 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+using FlagsmithEngine;
+using FlagsmithEngine.Environment.Models;
+using FlagsmithEngine.Feature.Models;
+using FlagsmithEngine.Identity.Models;
+using FlagsmithEngine.Segment;
+using FlagsmithEngine.Segment.Models;
+
+namespace Flagsmith
+{
+    /// <summary>
+    /// Utility class for transforming environment documents into evaluation contexts.
+    /// </summary>
+    public static class Mappers
+    {
+        /// <summary>
+        /// Property value for operators that do not use property (e.g., PERCENTAGE_SPLIT).
+        /// </summary>
+        private const string EmptyProperty = "";
+
+        /// <summary>
+        /// Key value for synthetic segments and identity override features.
+        /// </summary>
+        private const string UnusedKey = "";
+
+        /// <summary>
+        /// Parse the environment document into an EvaluationContext object
+        /// </summary>
+        public static EvaluationContext<SegmentMetadata, FeatureMetadata> MapEnvironmentDocumentToContext(
+            EnvironmentModel environmentDocument)
+        {
+            var context = new EvaluationContext<SegmentMetadata, FeatureMetadata>
+            {
+                Environment = new EnvironmentContext
+                {
+                    Key = environmentDocument.ApiKey,
+                    Name = environmentDocument.Project.Name,
```

Comment:
Indeed, the fixture is misleading and was missing the name (same happened in [Node](https://github.com/Flagsmith/flagsmith-nodejs-client/pull/226))